### ISSUE TITLE
[9.0, 10.0] pos: Fiscal Positions that remove taxes crashes the POS

### ISSUE THREAD

**Issue by gustavovalverde** [odoo/odoo#17963]
**Impacted versions:**
10.0

**Steps to reproduce:**
1 - Install POS
2 - Create a tax of 15%
3 - Create a Product A (PA) with this Tax
4 - Create a Fiscal Position (TaxRem) that removes this Tax and add it to the POS
5 - Open the POS, add a PA
6 - Apply the Fiscal Position TaxRem 

**Current behavior:**
Point of Sale crashes and it won't allow me to select any product with this tax, as it will also crash, this makes the Point of Sale completely useless.

Errrors:
```
Uncaught TypeError: Cannot read property 'id' of undefined
    at http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:195:451
    at Function._.each._.forEach (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:12:558)
    at _.(anonymous function) [as each] (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:69:526)
    at child.fix_tax_included_price (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:195:346)
    at child.add_product (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:199:6)
    at Class.click_product (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:338:257)
    at Object.click_product_action (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:337:951)
    at HTMLSpanElement.click_product_handler (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:334:1738)
```
AND

```
Uncaught TypeError: Cannot read property 'amount_type' of undefined
    at http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:174:7
    at Function._.each._.forEach (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:12:558)
    at _.(anonymous function) [as each] (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:69:526)
    at child.compute_all (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:173:135)
    at child.get_all_prices (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:178:590)
    at child.get_price_without_tax (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:164:898)
    at http://241873-10-0-0db908.runbot9.odoo.com/web/content/1900-d1ba930/point_of_sale.assets.js:208:446
    at iterator (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:14:183)
    at Function.<anonymous> (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1869-ad406d6/web.assets_common.js:17:8)
    at child.Collection.(anonymous function) [as reduce] (http://241873-10-0-0db908.runbot9.odoo.com/web/content/1881-9ab1124/web.assets_backend.js:1251:545)
```
**Configured and tested on runbot:** http://241873-10-0-0db908.runbot9.odoo.com

**Expected behavior:**
Removes the tax(es)

**Comment by gustavovalverde** [comment]
@odony @nim-odoo I'm adding you here as this happened to me after updating to the latest rev, and this commit might have caused this https://github.com/odoo/odoo/commit/7b494187e99f90359167bb02743bc2f4f5f81043

**Comment by Yenthe666** [comment]
Hey @gustavovalverde,

I've just tested this without any problem? My POS order:
![image](https://user-images.githubusercontent.com/6352350/27779901-0b53a3f8-5fc6-11e7-8208-6f27b8316ab6.png)

My product:
![image](https://user-images.githubusercontent.com/6352350/27779936-2c3ccdf6-5fc6-11e7-983a-91ba93d914f7.png)
Or did I miss something here?

**Comment by gustavovalverde** [comment]
Could you try it in runbot 
http://241873-10-0-0db908.runbot9.odoo.com, it still has my configuration:
![image](https://user-images.githubusercontent.com/16890942/27789654-5ee49406-5fbb-11e7-9e37-3a2dd27006d9.png)

Also, it seems this Fiscal Position configuration (without a tax on the right column is what's triggering this), check if your fiscal position is configure this way:
![image](https://user-images.githubusercontent.com/16890942/27789785-c03c4848-5fbb-11e7-8791-ba70e625e316.png)

**Comment by Yenthe666** [comment]
Hi @gustavovalverde,

Thanks for the extra information, this is indeed a bug.
@jorenvo Whenever you have some spare time, could you look at it? (not sure if I should still ping after labelling though 😉 )

**Comment by odony** [comment]
@gustavovalverde definitely not related to 7b49418, which only changes tests.

Also, the code that [breaks the POS](https://github.com/odoo/odoo/blob/b9131adb4832d47c8d715cbf38694ccc83e7021c/addons/point_of_sale/static/src/js/models.js#L1362) when you forget to set a target tax in the fiscal position mapping has been here since 358e285fd8ed75ac322213bd1d3382ff7fc28eba in 2015. It's reproducible in 9.0 too.

Therefore it seems unlikely that a recent update introduced it. Maybe you used to have a target tax in your mapping and you recently deleted it?

**Comment by gustavovalverde** [comment]
Oh, that's weird. I can swear this happened after the update. I'll test on an old instance.

**Comment by jorenvo** [comment]
Thank you for reporting the issue. It has been fixed in 9.0 at b5aaf0d527cdab794058ccbbf3cd5d215fd9a5af. This patch will be forward-ported to all supported later versions of Odoo.
